src/scripts/antithesis.zig

```diff
@@ -0,0 +1,241 @@
+//! Deployment script for our systest (src/testing/systest).
+//!
+//! * Builds the Java client and the associated workload using Maven
+//! * Builds Docker images for the workload, replicas, and config
+//! * Optionally pushes the images to the Antithesis registry
+//!
+//! Currently there's no support for triggering tests with this script.
+
+const std = @import("std");
+const builtin = @import("builtin");
+const log = std.log;
+const assert = std.debug.assert;
+
+const stdx = @import("../stdx.zig");
+const flags = @import("../flags.zig");
+const fatal = flags.fatal;
+const Shell = @import("../shell.zig");
+
+pub const CLIArgs = struct {
+    tag: []const u8,
+    push: bool = false,
+};
+
+const Image = enum { config, workload, replica };
+
+pub fn main(shell: *Shell, gpa: std.mem.Allocator, cli_args: CLIArgs) !void {
+    assert(try shell.exec_status_ok("docker --version", .{}));
+
+    // Docker tag to build and push
+    const tag = std.mem.trim(u8, cli_args.tag, &std.ascii.whitespace);
+    assert(tag.len > 0);
+
+    try shell.zig("build -Drelease", .{});
+
+    // Build Java client library
+    {
+        try shell.pushd("./src/clients/java");
+        defer shell.popd();
+
+        try shell.exec("mvn clean install --batch-mode --quiet -Dmaven.test.skip", .{});
+    }
+
+    // Build workload
+    {
+        try shell.pushd("./src/testing/systest/workload");
+        defer shell.popd();
+
+        try shell.exec("mvn clean package --batch-mode --quiet", .{});
+    }
+
+    const images = comptime std.enums.values(Image);
+    inline for (images) |image| {
+        try build_image(shell, gpa, image, tag);
+        if (cli_args.push) {
+            try push_image(shell, image, tag);
+        }
+    }
+}
+
+fn build_image(
+    shell: *Shell,
+    gpa: std.mem.Allocator,
+    comptime image: Image,
+    tag: []const u8,
+) !void {
+    switch (image) {
+        .config => {
+            const image_dir = try shell.exec_stdout("mktemp -d", .{});
```

Comment:
Rather than shelling out to mktemp, let's lift this piece of code to shell.zig:

https://github.com/tigerbeetle/tigerbeetle/pull/2260/files#diff-dee49dfa908b58ba4278b1d022ce0726c79da00b3a3a87d93fa0619e7836d46bR321


There's also `std.testing.TmpDir`, but it's awkwrard, as it doesn't expose the full path.